configure.ac

```diff
@@ -107,7 +107,9 @@ AS_IF([test "$enable_threads" != "no"], [
        fi
        AC_DEFINE([ENABLE_THREADS], [1], [Enable threads])
        AC_CHECK_HEADERS([stdatomic.h])
-
+       CHECK_GETTID(
+           [AC_DEFINE([HAVE_SYS_GETTID], [1], [Have support for SYS_gettid syscall])],
+           [AC_MSG_WARN([No support for SYS_gettid syscall])])
```

Comment:
We're currently not using `SYS_gettid` on Mac, and fortunately this warning isn't printed on OSX, because it's available (at least on Sierra/HighSierra) .  _Unfortunately_, I see that it's a deprecated symbol at compile time.  Perhaps I could try to clean this up using something like `AC_CANONICAL_HOST` and `$host_os`?